providers-sdk/v1/plugin/runtime.go

```diff
@@ -44,6 +44,9 @@ func (r *Runtime) ResourceFromRecording(name string, id string) (map[string]*llx
 	// recording and initialize them.
 	// TODO: we could use the provided information for a later request.
 	for k, v := range data.Fields {
+		if len(v.Data.Type) == 0 {
```

Comment:
without this we would land in IsArray, which would read typ[0], which would in turn panic